project/AkkaBuild.scala

```diff
@@ -100,14 +100,18 @@ object AkkaBuild {
       scalacOptions in Compile ++= (
         if (JavaVersion.isJdk8)
           Seq("-target:jvm-1.8")
-        else
+        else if (!CrossJava.Keys.useSystemJdk.value) {
           // -release 8 is not enough, for some reason we need the 8 rt.jar explicitly #25330
-          Seq("-release", "8", "-javabootclasspath", CrossJava.Keys.fullJavaHomes.value("8") + "/jre/lib/rt.jar")),
+          if (!CrossJava.Keys.fullJavaHomes.value.contains("8")) throw new IllegalStateException("Jdk8 is not installed - to build without it `set every akka.CrossJava.Keys.useSystemJdk := true`")
```

Comment:
Perhaps we should make it clear in the message that this means the resulting artifact will not work on JDK8?

Reply:
@raboof @nvollmar I just keep feeling like this is not the right approach because typically requirements on JVM versions are done in an `initialize` block, and I dislike the throwing of an exception. Might we consider in an if block check you've added something along the lines of this or similar?

```scala
initialize := {
   val _ = initialize.value // run the previous initialization
   // the logic
   assert(..., "The useful user message and JDK version requirement for this case")
}
```